src/app/pages/BridgeDepositPage/components/AmountSelector/index.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { bignumber } from 'mathjs';
+import styled from 'styled-components/macro';
+import type { Chain } from 'types';
+
+import { actions } from '../../slice';
+import { selectBridgeDepositPage } from '../../selectors';
+import { BridgeDictionary } from '../../dictionaries/bridge-dictionary';
+import { CrossBridgeAsset } from '../../types/cross-bridge-asset';
+import { useTokenBalance } from '../../hooks/useTokenBalance';
+import { AssetModel } from '../../types/asset-model';
+import { AmountInput } from '../AmountInput';
+import { FormGroup } from '../../../../components/Form/FormGroup';
+import { useBridgeLimits } from '../../hooks/useBridgeLimits';
+import { toNumberFormat } from '../../../../../utils/display-text/format';
+import { fromWei } from 'utils/blockchain/math-helpers';
+import { useBridgeTokenBalance } from '../../hooks/useBridgeTokenBalance';
+import { LoadableValue } from 'app/components/LoadableValue';
+import { ActionButton } from 'app/components/Form/ActionButton';
+import { translations } from 'locales/i18n';
+import { useTranslation } from 'react-i18next';
+
+export function AmountSelector() {
+  const { amount, chain, targetChain, sourceAsset, targetAsset } = useSelector(
+    selectBridgeDepositPage,
+  );
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const asset = useMemo(
+    () =>
+      BridgeDictionary.get(chain as Chain, targetChain)?.getAsset(
+        sourceAsset as CrossBridgeAsset,
+      ) as AssetModel,
+    [chain, sourceAsset, targetChain],
+  );
+
+  const [value, setValue] = useState(
+    amount ? asset.fromWei(amount, asset.minDecimals) : '',
+  );
+
+  const selectAmount = useCallback(() => {
+    dispatch(actions.selectAmount(asset.toWei(value || '0')));
+  }, [dispatch, asset, value]);
+
+  const balance = useTokenBalance(chain as any, asset);
+
+  const { value: limits, loading: limitsLoading } = useBridgeLimits(
```

Comment:
This is the same problem, you just cast it to `any` to bypass typescript but it doesn't solve the underlying problem, correct the typings.